tda/httpx_oauth2.py

```diff
@@ -0,0 +1,215 @@
+import asyncio
```

Comment:
I propose we move this to a "utils" or "third_party" directory, just to highlight the fact that it's a separate piece of code from the rest of the library? 

It would also be good to add a comment up top describing why this is here, complete with links to any PRs you have outstanding, etc.